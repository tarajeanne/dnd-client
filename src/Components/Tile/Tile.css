* {
   box-sizing: border-box;
}

h2 {
  font-family: 'Raleway', sans-serif;
}

p {
  font-family: 'Montserrat', sans-serif;
}

.option-tile {
  border: .5px solid #7A6263;
  padding: 5px;
  box-shadow: 0px 0px 8px .5px #7A6263;
  display: inline-block;
  margin: 8px;
  width: 308px;
  height: 308px;
  border-radius: 1px;
  z-index: 100;
}

.tile-radio {
  opacity:0;
}

.selected {
  box-shadow: 0px 0px 8px 3px #7EBDC3;
  border: .5px solid #7EBDC3;
}

