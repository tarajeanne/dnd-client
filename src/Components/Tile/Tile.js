import React from 'react';
import CharacterContext from '../../contexts/CharacterContext';
import './Tile.css';

class Tile extends React.Component {
  constructor(props){
    super(props);
    this.screen=this.props.screen;
  }
  static contextType = CharacterContext;
  handleClick = () => {
    this.props.handleClick(this.props.item.name)
  }
  render() {
    return (
      <div className={(this.props.selected == this.props.item.name) ? "option-tile selected" : "option-tile"} onClick={()=>{this.handleClick()}}>
        <header>
          <h2>{this.props.item.name}</h2>
        </header>
        <p className="option-description">{this.props.item.desc}</p>
      </div>
    );
  }
}

export default Tile;