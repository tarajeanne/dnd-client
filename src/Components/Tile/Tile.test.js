import React from 'react';
import Tile from './Tile';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Tile', () => {
  it('renders as expected', () => {
    const item = {
      id: 1,
      name: 'Dwarf',
      desc:
        'Proud warriors and miners from the mountains who value culture and tradition.',
      asi: [
        {
          name: 'Constitution',
          variable: false,
          depends_on: 'race',
          magnitude: 2
        }
      ],
      age:
        'Physically mature in their 20s, but not socially adults until 50. They live about 350 years.',
      alignment:
        'Believing in order, justice, and fair play, dwarves tend toward lawful good.',
      size: 'Medium',
      size_desc: 'Between 4 and 5 feet tall, averaging 150 lbs.',
      speed: 25,
      languages: [
        {
          name: 'Dwarvish',
          variable: false,
          depends_on: 'race'
        }
      ],
      weap_prof: [
        {
          weapid: 15,
          variable: false,
          depends_on: 'race'
        },
        {
          weapid: 21,
          variable: false,
          depends_on: 'race'
        },
        {
          weapid: 4,
          variable: false,
          depends_on: 'race'
        },
        {
          weapid: 6,
          variable: false,
          depends_on: 'race'
        }
      ],
      other_prof: [
        {
          name: '',
          variable: true,
          depends_on: 'race',
          options: ["smith's tools", "brewer's supplies", "mason's tools"]
        },
        {
          name: 'Stonecunning',
          desc:
            'Whenever you make a History check related to stonework, add double your proficiency bonus to the check.',
          depends_on: 'race',
          variable: false
        },
        {
          name: 'Dwarven Resilience',
          desc:
            'You have advantage on saving throws against poison, and you have resistance against poison damage',
          variable: false,
          depends_on: 'race'
        }
      ],
      skills_and_features: [
        {
          name: 'Darkvision',
          desc:
            'You can see in dim light within 60 feet of you as if it were bright light, and in darkness as if it were dim light.',
          variable: false,
          depends_on: 'race'
        }
      ],
      subraces: [
        {
          name: 'Hill Dwarf',
          slug: 'hill-dwarf',
          desc:
            'As a hill dwarf, you have keen senses, deep intuition, and remarkable resilience.',
          asi: {
            Wisdom: 1
          },
          asi_desc: 'Ability Score Increase. Your Wisdom score increases by 1',
          document_slug: 'systems-reference-document'
        }
      ]
    };
    const wrapper = shallow(<Tile item={item} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
