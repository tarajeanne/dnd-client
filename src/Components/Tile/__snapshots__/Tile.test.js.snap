// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders as expected 1`] = `
<label
  htmlFor="Dwarf-tile-select"
>
  <div
    className="option-tile"
  >
    <header>
      <h2>
        Dwarf
      </h2>
    </header>
    <input
      checked={false}
      className="tile-radio"
      id="Dwarf-tile-select"
      onChange={[Function]}
      type="radio"
      value="Dwarf"
    />
    <p
      className="option-description"
    >
      Proud warriors and miners from the mountains who value culture and tradition.
    </p>
  </div>
</label>
`;
